### PR TITLE
`CellariumPipeline`

### PR DESCRIPTION
**Pull request by ordabayevy**
- Added `core.CellariumPipeline` to pipe transforms and the model
- Changed `g_genes` to `feature_schema` in `IncrementalPCA`, `OnePassMeanVarStd`, and `TDigest`
- Removed `from __future__ import annotations` (docs render better without it)
- Refactored transforms out of the models and removed all `*FromCLI` models
- Removed `_get_fn_args_from_batch` from the models
- Models and transforms return a dict now
- Fix issues in the documentation building

### CODE REVIEW

File: cellarium/ml/models/incremental_pca.py [hunk 1]
```diff
@@ -36,24 +40,22 @@ class IncrementalPCA(CellariumModel, PredictMixin):
...
-        g_genes: int,
+        feature_schema: Sequence[str],
         k_components: int,
         svd_lowrank_niter: int = 2,
         perform_mean_correction: bool = False,
-        transform: nn.Module | None = None,
     ) -> None:
         super().__init__()
+        self.feature_schema = np.array(feature_schema)
+        g_genes = len(self.feature_schema)
```
Comment (mbabadi) [inline cellarium/ml/models/incremental_pca.py:54]: Please rename `g_genes` here and elsewhere to `n_features`.

File: cellarium/ml/models/incremental_pca.py [hunk 2]
```diff
@@ -26,8 +30,8 @@ class IncrementalPCA(CellariumModel, PredictMixin):
        <https://www.cs.toronto.edu/~dross/ivt/RossLimLinYang_ijcv.pdf>`_.
 
     Args:
-        g_genes:
-            Number of genes.
+        feature_schema:
```
Comment (mbabadi) [inline cellarium/ml/models/incremental_pca.py:33]: Please rename `feature_schema` here and elsewere to `feature_names`. The word "schema" is reserved for a greater scope of data structuring and I am anticipating that we will run into name collisions in the future.

**Review by mbabadi — COMMENTED** [review]
As a general comment, I would love to see a convergence between `feature_g` and `feature_schema` to the same thing, perhaps `feature_names_g` or better yet, `var_names_g`.

File: cellarium/ml/transforms/filter.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ class Filter(nn.Module):
         filter_list: A list of features to filter by.
     """
 
-    def __init__(self, filter_list: ArrayLike) -> None:
+    def __init__(self, filter_list: Sequence[str]) -> None:
```
Comment (mbabadi) [inline cellarium/ml/transforms/filter.py:31]: Please add docstring to `Filter` that `feature_g` will be overwritten.

File: cellarium/ml/transforms/log1p.py [hunk 4]
```diff
@@ -18,15 +18,18 @@ class Log1p(nn.Module):
     def __init__(self) -> None:
         super().__init__()
 
-    def forward(self, x_ng: torch.Tensor) -> torch.Tensor:
+    def forward(self, x_ng: torch.Tensor) -> dict[str, torch.Tensor]:
```
Comment (mbabadi) [inline cellarium/ml/transforms/log1p.py:21]: Add docstring that the operation is in-plane.

File: cellarium/ml/transforms/normalize_total.py [hunk 5]
```diff
@@ -43,7 +43,7 @@ def forward(
         self,
         x_ng: torch.Tensor,
         total_mrna_umis_n: torch.Tensor | None = None,
-    ) -> torch.Tensor:
+    ) -> dict[str, torch.Tensor]:
```
Comment (mbabadi) [inline cellarium/ml/transforms/normalize_total.py:46]: Add docstring in the class doc that the operation is in-place.

File: cellarium/ml/transforms/z_score.py [hunk 6]
```diff
@@ -62,13 +62,15 @@ def forward(
                 The list of the variable names in the input data. If ``None``, no validation is performed.
 
         Returns:
-            Z-scored gene counts.
+            A dictionary with the following keys:
+
+            - ``x_ng``: The z-scored gene counts.
```
Comment (mbabadi) [inline cellarium/ml/transforms/z_score.py:67]: Add class docstring that the operation is in-place.

**Review by mbabadi — COMMENTED** [review]
General comment: none of the tests in `tests/test_cli.py` have `transforms`.  I suggest adding `transforms` to `onepass_mean_var_std` and `incremental_pca` tests.

**Review by mbabadi — COMMENTED** [review]
General comment: please make an `examples` directory (or somthing to that effect) and populate it with several useful config files, each with a separate README. For example:
- OnePassMeanVarStd -> Incremental PCA
- TDigest w/ lib size normalization + log transform -> LogisticRegression w/ lib size normalization + log transform + divide by median

**Review by mbabadi — COMMENTED** [review]
General comment: please make an issue to make an architecture diagram of the codebase: how the code is organized, the class hierarchy, etc.

**Review by mbabadi — CHANGES_REQUESTED** [review]
Nice work! :)

File: cellarium/ml/callbacks/prediction_writer.py [hunk 7]
```diff
@@ -65,18 +65,19 @@ def write_on_batch_end(
         self,
         trainer: pl.Trainer,
         pl_module: pl.LightningModule,
-        prediction: torch.Tensor,
+        prediction: dict[str, torch.Tensor],
```
[the remark this answers is not in the document]
Comment (ordabayevy) [reply cellarium/ml/callbacks/prediction_writer.py:68]: Created a separate issue for this since it might require some additional validation

**Review by mbabadi — CHANGES_REQUESTED** [review]
All good, let's merge! :)

**Review by mbabadi — APPROVED** [review]
